packages/x-charts/src/internals/store/useCharts.ts

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import { ChartStore } from '../plugins/utils/ChartStore';
+import {
+  ChartAnyPluginSignature,
+  ChartInstance,
+  ChartPlugin,
+  ChartPublicAPI,
+  ChartState,
+  ConvertSignaturesIntoPlugins,
+} from '../plugins/models';
+import { CHART_CORE_PLUGINS, ChartCorePluginSignatures } from '../plugins/corePlugins';
+import { UseChartBaseProps } from './useCharts.types';
+import { UseChartInteractionState } from '../plugins/featurePlugins/useChartInteraction/useChartInteraction.types';
+
+export function useChartApiInitialization<T>(
+  inputApiRef: React.MutableRefObject<T | undefined> | undefined,
+): T {
+  const fallbackPublicApiRef = React.useRef({}) as React.MutableRefObject<T>;
+
+  if (inputApiRef) {
+    if (inputApiRef.current == null) {
+      // eslint-disable-next-line react-compiler/react-compiler
```

Comment:
That's because eslint does not see we are using a ref

Reply:
Should work once we have https://github.com/facebook/react/pull/29916